lib/round.rb

```diff
@@ -0,0 +1,47 @@
+class Round
+    attr_reader :deck
+    attr_reader :turns
+
+    def initialize(deck)
+        @deck = deck
+        @turns = []
+        @number_correct = 0
+    end
+
+
+    def current_card 
+        @deck.cards[@turns.count]
+    end
+
+    def take_turn(guess)
+        new_turn = Turn.new(guess, current_card)
+        @turns << new_turn
+        if new_turn.correct?
+            @number_correct += 1
+        end
+        new_turn
+    end
+    
+    def number_correct
+        @number_correct
+    end
+    
+    def number_correct_by_category(category)
+        @turns.count do |turn|
+            turn.card.answer == turn.guess && turn.card.category == category
```

Comment:
You already have a method in the `Turn` class called `correct?` that you can lean on for the first half of this condition. `turn.correct? && turn.card.category == category`